### PR TITLE
Record Meter: don't use last_indexed_date to decide if a site is indexed or not

### PR DESCRIPTION
**Pull request by bluefuton**
We currently rely on the `last_indexed_date` from the API to decide if a site is indexed or not. During the recent call-for-testing, @jeherve showed us an API response for a site that clearly had records, but `last_indexed_date` was null.

image

#### Changes proposed in this Pull Request:
This PR removes `last_indexed_date` from our logic when deciding if a site has been indexed.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion
p8oabR-Tl-p2#comment-6400

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
It's hard to recreate this situation without manually changing the API response, but you can check that the sign up and indexing process work as expected:

1. Create a new test site.
2. Install the Jetpack Search plugin.
3. Visit `/wp-admin/admin.php?page=jetpack-search&features=record-meter` and check that you're shown appropriate messages during the indexing process.

You can also ensure that the unit tests pass.

`cd projects/packages/search && pnpm test`

### CODE REVIEW

File: projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ export default function getRecordInfo( postCount, postTypeBreakdown, lastIndexed
 			postTypes
 		);
 
-		for ( let i = 0; i < numItems - 1; i++ ) {
+		for ( let i = 0; i < numItems; i++ ) {
```
Comment (bluefuton) [inline projects/packages/search/src/dashboard/components/record-meter/lib/record-info.js:55]: `addLabelsToPostTypeBreakdown` hands us back a zero-indexed array, so I believe we don't need the `-1`. This was resulting in the final post type never being displayed.

**Review by jeherve — DISMISSED** [review]
This tests well on the site where I experienced the issue before.

image

**Review by kangzj — DISMISSED** [review]
LGTM

**Comment by bluefuton** [comment]
> This tests well on the site where I experienced the issue before.

Thanks @jeherve! Can I just check that the number formatting is correct on the legend in your screenshot? Looks like perhaps missing a comma?

Screen Shot 2022-07-07 at 10 32 59

**Review by kangzj — APPROVED** [review]

**Comment by a8ck3n** [comment]
> Can I just check that the number formatting is correct on the legend in your screenshot? Looks like perhaps missing a comma?

I think the formatting there is correct. I believe the formatting should be "12 123,99". So it appears to be wrong in the messaging above the Record Meter bar.

**Comment by a8ck3n** [comment]
Just confirmed the above. Currently we use `Number.toLocaleString()` to prepare the summary message. In the Legend, we are using `numberFormat()` which seems to do a better job of detecting the current locale and gives us the expected output.

I'll open a new issue for this.

**Comment by jeherve** [comment]
I think it may be a quirk of my setup, sorry y'all. My test site is in French, but my browser is set to English, so whether we use the browser or the site locale to determine language, we get different formats.

I don't know if that's a big problem, I can't imagine many site owners are in that situation?

**Comment by bluefuton** [comment]
> Currently we use Number.toLocaleString() to prepare the summary message. In the Legend, we are using numberFormat() which seems to do a better job of detecting the current locale and gives us the expected output.

I think the important thing is to choose one method and be consistent, so we have the same formatting throughout the meter.

> I'll open a new issue for this.

Thanks @a8ck3n! Could you link it here when you've created it?

**Comment by a8ck3n** [comment]
I've created the GH Issue here:

https://github.com/Automattic/jetpack/issues/25004

I have the required changes in a Draft PR here:

https://github.com/Automattic/jetpack/pull/25003

Once I get my test site working (the React Dev tools issue mentioned elsewhere), I'll come back to that PR and get it filled out properly.

**Pull request merged**